File: app/src/main/java/br/svcdev/githubclient/MainActivity.java
package br.svcdev.githubclient;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import br.svcdev.githubclient.databinding.ActivityMainBinding;
import br.svcdev.githubclient.presenter.Presenter;
import br.svcdev.githubclient.view.MainView;

public class MainActivity extends AppCompatActivity implements MainView, View.OnClickListener {

	private Presenter presenter;
	private ActivityMainBinding mBinding;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		mBinding = ActivityMainBinding.inflate(getLayoutInflater());
		setContentView(mBinding.getRoot());

		init();
	}

	private void init() {
		presenter = new Presenter(this);

		mBinding.btnCounter1.setTag(0);
		mBinding.btnCounter2.setTag(1);
		mBinding.btnCounter3.setTag(2);

		mBinding.btnCounter1.setOnClickListener(this);
		mBinding.btnCounter2.setOnClickListener(this);
		mBinding.btnCounter3.setOnClickListener(this);
	}

	@Override
	public void setButtonText(int index, String text) {
		Button button = mBinding.getRoot().findViewWithTag(index);
		button.setText(text);
	}

	@Override
	public void onClick(View view) {
		presenter.onCounterClick((int) view.getTag());
	}
}

File: app/src/main/java/br/svcdev/githubclient/MainActivity.kt
package br.svcdev.githubclient

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import br.svcdev.githubclient.databinding.ActivityMainBinding
import br.svcdev.githubclient.presenter.Presenter
import br.svcdev.githubclient.view.MainView

class MainActivity : AppCompatActivity(), MainView, View.OnClickListener {
    private val presenter = Presenter(this)
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        init()
    }

    private fun init() {
        binding.btnCounter1.setOnClickListener(this)
        binding.btnCounter2.setOnClickListener(this)
        binding.btnCounter3.setOnClickListener(this)
    }

    override fun setButtonOneText(text: String?) {
        binding.btnCounter1.text = text
    }

    override fun setButtonTwoText(text: String?) {
        binding.btnCounter2.text = text
    }

    override fun setButtonThreeText(text: String?) {
        binding.btnCounter3.text = text
    }

    override fun onClick(view: View) {
        when(view){
            binding.btnCounter1 -> presenter.onButtonOneClick()
            binding.btnCounter2 -> presenter.onButtonTwoClick()
            binding.btnCounter3 -> presenter.onButtonThreeClick()
        }
    }
}